templates/common/_base/files/nmstate-configuration.yaml

```diff
@@ -0,0 +1,42 @@
+mode: 0755
+path: "/usr/local/bin/nmstate-configuration.sh"
+contents:
+  inline: |
+    #!/bin/bash
+    set -eux
+
+    # Clean up old config on behalf of mtu-migration
+    if ! systemctl -q is-enabled mtu-migration; then
+      echo "Cleaning up left over mtu migration configuration"
+      rm -rf /etc/cno/mtu-migration
+    fi
+
+    if [ -e /etc/nmstate/openshift/applied ]; then
+      # Already done, don't apply again
+      exit 0
+    fi
+
+    src_path="/etc/nmstate/openshift"
+    dst_path="/etc/nmstate"
+    hostname=$(hostname -s)
+    hostname_file="${hostname}.yml"
+    cluster_file="cluster.yml"
+    config_file=""
+    if [ -s "$src_path/$hostname_file" ]; then
+      config_file=$hostname_file
+    fi
+    if [ -s "$src_path/$cluster_file" ]; then
```

Comment:
hi @cybertron if i understand correctly, hostname_file is node-specific configuration file and cluster_file is cluster-wide configuration file, right? if so, seems the cluster-wide one will take precedence, does it conflict with the description in enhancement?

Reply:
You're right. This should be fixed by the change in logic in the next version of this.